axelor-stock/src/main/java/com/axelor/apps/stock/service/StockLocationLineServiceImpl.java

```diff
@@ -98,37 +98,40 @@ public void updateLocation(
       BigDecimal reservedQty)
       throws AxelorException {
 
-    StockLocationLine stockLocationLine = this.getOrCreateStockLocationLine(stockLocation, product);
-
-    LOG.debug(
-        "Mise à jour du stock : Entrepot? {}, Produit? {}, Quantité? {}, Actuel? {}, Futur? {}, Incrément? {}, Date? {}, Num de suivi? {} ",
-        stockLocation.getName(),
-        product.getCode(),
-        qty,
-        current,
-        future,
-        isIncrement,
-        lastFutureStockMoveDate);
+    if (!product.getStockManaged()) {
+      StockLocationLine stockLocationLine =
+          this.getOrCreateStockLocationLine(stockLocation, product);
+
```

Comment:
move product.getStockManaged() int to getOrCreateStockLocationLine method, and put null check everywhere where that method is called.